configure: fix static link with alsa

Use pkg-config to find the alsa library, that will return all the
dependent libraries of alsa (e.g. -ldl -lpthread -lrt) as well that
can not be deduced in case of a static link.

Signed-off-by: "Yann E. MORIN" <yann.morin.1998@free.fr>

---
Patch applied upstream:
    https://github.com/mikebrady/shairport-sync/commit/e5a74a3ff08134a2b11f07bd850659d7816fb03d

diff -durN shairport-sync-2.1.8.orig/configure.ac shairport-sync-2.1.8/configure.ac
--- shairport-sync-2.1.8.orig/configure.ac	2014-11-29 12:56:11.954118681 +0100
+++ shairport-sync-2.1.8/configure.ac	2014-11-29 13:07:24.941687600 +0100
@@ -99,7 +99,10 @@
   HAS_ALSA=1
   AM_CONDITIONAL([USE_ALSA], [test 0])
   AC_DEFINE([CONFIG_ALSA], 1, [Needed by the compiler.])
-  AC_CHECK_LIB([asound], [snd_pcm_open], , AC_MSG_ERROR(ALSA support requires the asound library!))], )
+  PKG_CHECK_MODULES(
+    [ALSA], [alsa],
+    [LIBS="${ALSA_LIBS} ${LIBS}"
+     AC_DEFINE([HAVE_LIBASOUND],[1],[Define to 1 if you have ALSA])])])
 AM_CONDITIONAL([USE_ALSA], [test "x$HAS_ALSA" = "x1"])
 
 # Look for SNDIO flag
